### DISCUSSION TITLE
When RAILS_ENV=test Read-only file system @ rb_sysopen - /app/tmp/development_secret.txt"

### DISCUSSION THREAD

**Discussion by michaelcohenunsw** [customink/lambdakiq#20] [Q&A]
Hi,

This isn't specifically a lambdakiq issue. It is more of a non-production Rails annoyance. We are using RAILS_ENV=test and FUNCTION_DIR=/app in our AWS Lambda docker image. 

The lambdakiq background job for sending mail fails with error: Read-only file system @ rb_sysopen - /app/tmp/development_secret.txt". 

However, lambdakiq background jobs for promoting and demoting Shrine file uploads work fine. Is the difference here because eager loading = false for RAILS_ENV=test? Would both generally fail when RAILS_ENV=test or is there a workaround possible without renaming the RAILS_ENV to something other than "test"?

**Comment by metaskills** [comment]
I have no idea what might be trying to write to that file. Would this be as simple as tracking down the code that writes to that file and making that one fix to either not write the file or to write it to /tmp?
